src/register.js

```diff
@@ -107,16 +108,11 @@ var RouteRegistry = {
 			if (this.data instanceof RouteData) {
 				var routeData = this.data;
 				var definePropertyWithDefault = function(name) {
-					var type = "string",
-						defaultValue = defaults[name],
-						typeOf = typeof defaultValue;
-
-					if (defaultValue != null) {
-						type = typeOf;
-					}
+					var defaultValue = defaults[name];
+					var propertyType = defaultValue != null ? type.maybeConvert(defaultValue.constructor) : type.maybeConvert(String);
```

Comment:
@phillipskevin we had talked about this handling both `can-define/map/map` and `can-observable-object`, but I don’t think we need to. This logic was added for when no `data` is specified, not generically for any `can-define/map/map`, so now that you have to provide `data` if you want `can-define/map/map`, I don’t think this logic needs to support it.